dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/model/fields/SearchKey.java

```diff
@@ -0,0 +1,118 @@
+package org.icgc.dcc.portal.server.model.fields;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.val;
+import org.icgc.dcc.common.core.util.Joiners;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Sets.newHashSet;
+import static org.icgc.dcc.common.core.util.stream.Collectors.toImmutableSet;
+import static org.icgc.dcc.portal.server.model.fields.SearchField.newBoostedSearchField;
+
+/**
+ * Represents a key term, which contain subfields. The class can generate SearchFields RELATIVE to the key name, using a collection of previously defined SearchFields. The purpose of this class is to partially model the Elasticsearch mapping below:
```

Comment:
adds this to the docs here: https://github.com/icgc-dcc/dcc-portal/blob/develop/dcc-portal-server/SERVER.md